ldap/ldap.go

```diff
@@ -107,6 +114,13 @@ func (opts CreationOpts) validate() error {
 			opts.URL, opts.Port, opts.UserPath, opts.ServicePath))
 	}
 
+	if opts.ServiceUserName == "" {
+		catcher.Add(errors.New("LDAP service user name cannot be empty"))
+	}
+	if opts.ServiceUserPassword == "" {
+		catcher.Add(errors.New("LDAP service user password cannot be empty"))
```

Comment:
you can do the catcher.NewWhen() to clarify all of these calls

Reply:
Done.